internal/controllers/machinedeployment/machinedeployment_sync.go

```diff
@@ -254,6 +262,42 @@ func (r *Reconciler) getNewMachineSet(ctx context.Context, d *clusterv1.MachineD
 	return createdMS, err
 }
 
+func generateMachineSetName(d *clusterv1.MachineDeployment, now *metav1.Time) (string, string, error) {
```

Comment:
Hash is currently used as:
- UID to identify machines belonging to the MS
- Set a suffix to the MS set name

Given that I'm really wondering if we should drop the current spew/hash logic and simply use a random number + a check that verified the random number is not taken. It seems that the code could be re-entrant also it this way

@vincepri @enxebre @sbueringer opinions?

Reply:
@fabriziopandini  Sorry missed the mention somehow.

Sounds fine to me, assuming we can make this re-entrant (I didn't look at the code in detail to see how this would be achieved).

+100 to making this a separate PR independent of this work and the propagation work

Would be nice to get rid of the hash early in the v1.4 cycle to give us a lot of time to discover potential side effects